File: example/src/main/java/com/attentive/example/activities/MainActivity.java
package com.attentive.example.activities;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.EditText;
import androidx.appcompat.app.AppCompatActivity;
import com.attentive.example.ExampleApp;
import com.attentive.example.R;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setupDomainEditText();
    }

    public void startLoadCreativeActivity(View view) {
        Intent intent = new Intent(this, LoadCreativeActivity.class);
        startActivity(intent);
    }

    public void startProductPageActivity(View view) {
        Intent intent = new Intent(this, ProductPageActivity.class);
        startActivity(intent);
    }

    public void logoutUser(View view) {
        // Perform app's normal logout functionality

        // Clear all Attentive identifiers
        ((ExampleApp) this.getApplication()).getAttentiveConfig().clearUser();
    }

    public void identifyUser(View view) {
        ((ExampleApp) this.getApplication()).getAttentiveConfig().identify(ExampleApp.buildUserIdentifiers());
    }

    private void setupDomainEditText() {
        EditText domainValueTextView = findViewById(R.id.domainValue);
        domainValueTextView.setText(((ExampleApp) this.getApplication()).getAttentiveConfig().getDomain());
        domainValueTextView.setOnEditorActionListener((v, actionId, event) -> {
            if (actionId == EditorInfo.IME_ACTION_DONE) {
                ((ExampleApp) getApplication()).getAttentiveConfig().changeDomain(domainValueTextView.getText().toString());
                return true;
            }
            return false;
        });
    }
}

File: example-kotlin/src/main/java/com/attentive/example_kotlin/activities/MainActivity.kt
package com.attentive.example_kotlin.activities

import android.content.Intent
import android.os.Bundle
import android.view.KeyEvent
import android.view.View
import android.view.inputmethod.EditorInfo
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.attentive.example_kotlin.ExampleKotlinApp
import com.example.example_kotlin.R
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setupDomainEditText()
    }

    fun startLoadCreativeActivity(view: View?) {
        val intent = Intent(this, LoadCreativeActivity::class.java)
        startActivity(intent)
    }

    fun startProductPageActivity(view: View?) {
        val intent = Intent(this, ProductPageActivity::class.java)
        startActivity(intent)
    }

    fun logoutUser(view: View?) {
        // Perform app's normal logout functionality

        // Clear all Attentive identifiers
        (this.application as ExampleKotlinApp).attentiveConfig.clearUser()
    }

    fun identifyUser(view: View?) {
        (this.application as ExampleKotlinApp).attentiveConfig.identify(ExampleKotlinApp.buildUserIdentifiers())
    }

    private fun setupDomainEditText() {
        val domainValueTextView = findViewById<EditText>(R.id.domainValue)
        domainValueTextView.setText((this.application as ExampleKotlinApp).attentiveConfig.domain)
        domainValueTextView.setOnEditorActionListener { _: TextView?, actionId: Int, _: KeyEvent? ->
            if (actionId == EditorInfo.IME_ACTION_DONE) {
                (this.application as ExampleKotlinApp).attentiveConfig.changeDomain(domainValueTextView.text.toString())
            }
            return@setOnEditorActionListener false
        }
    }
}